pkgs/build-support/cc-wrapper/add-hardening.sh

```diff
@@ -34,6 +34,16 @@ if (( "${NIX_DEBUG:-0}" >= 1 )); then
   fi
 fi
 
+# Ensure PIE flags occur before PIC, if applicable
+if [ "${hardeningEnableMap[pie]-}" = 1 ]; then
+  if (( "${NIX_DEBUG:-0}" >= 1 )); then echo HARDENING: enabling CFlags -fPIE >&2; fi
+  hardeningCFlags+=('-fPIE')
```

Comment:
Can we instead do
`hardeningCFlags =( '-fPIE' "${hardeningCFlags[@]}" )`

Reply:
Great idea, that is SO much cleaner than what I have. Gonna run the rebuild gauntlet on this and get back to you.

Reply:
Works great!